### PR TITLE
Rancher 2.5 control plane docs

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?
Document how to configure Control Plane integrations on Rancher RKE 2.5

### Preview

https://docs-staging.datadoghq.com/ahmed/rancher-control-plane/agent/kubernetes/control_plane

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/agent/kubernetes/control_plane.md [hunk 1]
```diff
@@ -598,6 +599,181 @@ The Datadog Cluster Agent schedules the checks as endpoint checks and dispatches
...
+metadata:
+  name: datadog
+spec:
+  credentials:
+    apiKey: <DATADOG_API_KEY>
+    appKey: <DATADOG_APP_KEY>
+  clusterName: <CLUSTER_NAME>
+  agent:
+    image:
+      name: "gcr.io/datadoghq/agent:latest"
```
Comment (clamoriniere) [inline content/en/agent/kubernetes/control_plane.md:748]: thanks to operator v0.7.0 and the defaulting logic. I think we can remove the need of defining the image

```suggestion
```

File: content/en/agent/kubernetes/control_plane.md [hunk 2]
```diff
@@ -598,6 +599,181 @@ The Datadog Cluster Agent schedules the checks as endpoint checks and dispatches
...
+      tolerations:
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/controlplane
+          operator: Exists
+        - effect: NoExecute
+          key: node-role.kubernetes.io/etcd
+          operator: Exists
+  clusterAgent:
+    image:
+      name: "gcr.io/datadoghq/cluster-agent:latest"
```
Comment (clamoriniere) [inline content/en/agent/kubernetes/control_plane.md:769]: same for the `clusterAgent.image`

```suggestion
```

**Review by clamoriniere — COMMENTED** [review]
Looks great !
just a view comment. 
quick question: no additional RBAC was required?

**Review by clamoriniere — APPROVED** [review]

File: content/en/agent/kubernetes/control_plane.md [hunk 3]
```diff
@@ -598,6 +599,177 @@ The Datadog Cluster Agent schedules the checks as endpoint checks and dispatches
...
 On other managed services, such as Azure Kubernetes Service (AKS) and Google Kubernetes Engine (GKE), the user cannot access the control plane components. As a result, it is not possible to run the `kube_apiserver`, `kube_controller_manager`, `kube_scheduler`, and `etcd` checks in these environments.
 
+## Kubernetes on Rancher Kubernetes Engine (v2.5+) {#RKE}
+
+Rancher v2.5 relies on [PushProx][10] to expose control plane metric endpoints, this allows the Datadog Agent to run control plane checks and collect metrics.
+
+### Prerequisites
+
+1. Install the [rancher-monitoring chart][11]
+1. The `pushprox` daemonsets are deployed via `rancher-monitoring` and running in the `cattle-monitoring-system` namespace.
```
Comment (jtappa) [inline content/en/agent/kubernetes/control_plane.md:609]: Our flavor of markdown gets buggy if you leave numbered lists all the same number 😢 
```suggestion
1. Install the [rancher-monitoring chart][11].
2. The `pushprox` daemonsets are deployed with `rancher-monitoring` and running in the `cattle-monitoring-system` namespace.
```

File: content/en/agent/kubernetes/control_plane.md [hunk 4]
```diff
@@ -598,6 +599,177 @@ The Datadog Cluster Agent schedules the checks as endpoint checks and dispatches
...
+```yaml
+annotations:
+  ad.datadoghq.com/endpoints.check_names: '["kube_apiserver_metrics"]'
+  ad.datadoghq.com/endpoints.init_configs: '[{}]'
+  ad.datadoghq.com/endpoints.instances: '[{ "prometheus_url": "https://%%host%%:%%port%%/metrics", "bearer_token_auth": "true" }]'
+```
+
+### ETCD, Controller Manager and Scheduler
+
+By adding headless Kubernetes services to define check configurations, the Datadog Agent will be able to target the `pushprox` pods and collect metrics.
```
Comment (jtappa) [inline content/en/agent/kubernetes/control_plane.md:624]:
```suggestion
By adding headless Kubernetes services to define check configurations, the Datadog Agent is able to target the `pushprox` pods and collect metrics.
```

File: content/en/agent/kubernetes/control_plane.md [hunk 5]
```diff
@@ -598,6 +599,177 @@ The Datadog Cluster Agent schedules the checks as endpoint checks and dispatches
...
+## Kubernetes on Rancher Kubernetes Engine (v2.5+) {#RKE}
+
+Rancher v2.5 relies on [PushProx][10] to expose control plane metric endpoints, this allows the Datadog Agent to run control plane checks and collect metrics.
+
+### Prerequisites
+
+1. Install the [rancher-monitoring chart][11]
+1. The `pushprox` daemonsets are deployed via `rancher-monitoring` and running in the `cattle-monitoring-system` namespace.
+
+### API Server
```
Comment (jtappa) [inline content/en/agent/kubernetes/control_plane.md:611]:
```suggestion
### API server
```

File: content/en/agent/kubernetes/control_plane.md [hunk 6]
```diff
@@ -598,6 +599,177 @@ The Datadog Cluster Agent schedules the checks as endpoint checks and dispatches
...
+To configure the `kube_apiserver_metrics` check, add the following annotations to the `default/kubernetes` service:
+
+```yaml
+annotations:
+  ad.datadoghq.com/endpoints.check_names: '["kube_apiserver_metrics"]'
+  ad.datadoghq.com/endpoints.init_configs: '[{}]'
+  ad.datadoghq.com/endpoints.instances: '[{ "prometheus_url": "https://%%host%%:%%port%%/metrics", "bearer_token_auth": "true" }]'
+```
+
+### ETCD, Controller Manager and Scheduler
```
Comment (jtappa) [inline content/en/agent/kubernetes/control_plane.md:622]:
```suggestion
### Add Kubernetes services
```

**Review by jtappa — CHANGES_REQUESTED** [review]

File: content/en/agent/kubernetes/control_plane.md (hunk 6 above)
Comment (clamoriniere) [reply content/en/agent/kubernetes/control_plane.md:622]: @jtappa can we be more specific like
```suggestion
### Add Kubernetes services to configure auto-discovery checks
```

Comment (jtappa) [reply content/en/agent/kubernetes/control_plane.md:622]: Yes, that's better!

**Comment by clamoriniere** [comment]
@jtappa let me know if you see other change needed

**Review by jtappa — APPROVED** [review]

**Pull request merged**